raiden/api/rest.py

```diff
@@ -718,20 +733,23 @@ def get_connection_managers_info(self, registry_address: typing.TokenNetworkRegi
             registry_address=to_checksum_address(registry_address),
         )
         connection_managers = dict()
+        raiden_service = self.raiden_api.raiden
 
         for token in self.raiden_api.get_tokens_list(registry_address):
             token_network_address = views.get_token_network_address_by_token_address(
                 views.state_from_raiden(self.raiden_api.raiden),
                 token_network_registry_address=registry_address,
                 token_address=token,
             )
+            connection_manager = None
 
-            try:
-                connection_manager = self.raiden_api.raiden.connection_manager_for_token_network(
-                    token_network_address
-                )
-            except InvalidBinaryAddress:
-                connection_manager = None
+            if token_network_address is not None:
```

Comment:
I am assuming that this is a solution to one of the two bugs you mentioned in the description. Can't spot the other one though.

Reply:
You did, it is in your previous comment: https://github.com/raiden-network/raiden/pull/4724#discussion_r320187088 , the value `None` is invalid.